storybookjs/storybook
# Enable groupId option for knobs to be organized into sub panels.

Issue: #1896 

## What I did

- Added groupId argument to knob functions
- Created GroupTabs component (replica of AddonPanel in storybook/ui) within Knobs panel to serve as navigation when filtering between groups

## How to test
`groupId` has been added as the last argument to each of the knob functions (see updates to README)
```
text(label, defaultValue, groupId);
boolean(label, defaultValue, groupId);
number(label, defaultValue, options, groupId);
color(label, defaultValue, groupId);
object(label, defaultValue, groupId);
array(label, defaultValue, separator, groupId);
select(label, options, defaultValue, groupId);
date(label, defaultValue, groupId);
button(label, handler, groupId);
```

**NOTE: Knobs with optional arguments MUST have default values passed in for use with groupId.**

This was done to avoid breaking changes to current knob function contracts.

e.g. Notice these README updates:
- number(label, defaultValue, options, groupId);
- array(label, defaultValue, separator, groupId);

---

Is this testable with jest or storyshots?

Does this need a new example in the kitchen sink apps?

- Example usage has been provided in official-storybook app.

Does this need an update to the documentation?

- README has been updated with new argument.

If your answer is yes to any of these, please make sure to include it in your PR.

  
  
  
  
<img width="1437" alt="official-storybook" src="https://user-images.githubusercontent.com/4038373/34627165-881337e2-f22d-11e7-8aa4-b222b45b2d22.png">
